packages/origin-247-certificate/src/offchain-certificate/offchain-certificate.service.ts

```diff
@@ -75,35 +76,50 @@ export class OffchainCertificateService<T = null> {
         return certs;
     }
 
-    public async batchClaim(command: IClaimCommand[]): Promise<void> {
-        await Promise.all(
-            command.map(async (c: IClaimCommand) => {
-                await this.claim(c);
-            })
-        );
+    public async batchClaim(commands: IClaimCommand[]): Promise<void> {
+        for (const command of commands) {
+            await this.claim(command);
```

Comment:
I guess it would be better to optimize that: by applying multiple events on the same aggregate.
Also this implementation is wrong, because in error case whole batch should fail, and here you will apply first N commands, and fail afterwards.

Also in this implementation you rebuild aggregate unnecessary after each command.

Reply:
Why should a whole batch fail in that case? I could agree with failing all commands that concern the same certificateId, but not the whole batch when one command fails